nx/lib/nx/defn/expr.ex

```diff
@@ -213,7 +213,14 @@ defmodule Nx.Defn.Expr do
   """
   def while(initial, context, arg, condition, body) do
     [flatten_initial, flatten_arg, flatten_body] = clauses = flatten_clauses([initial, arg, body])
-    args = [flatten_initial, flatten_arg, condition, flatten_body]
+
+    args = [
+      Nx.devectorize(flatten_initial, keep_names: false),
+      flatten_arg,
+      Nx.devectorize(condition, keep_names: false),
```

Comment:
We should raise if the condition is vectorized. The same for the cond condition. We should raise if they are vectorized because the correct is to handle them properly.